test/format/util.c

```diff
@@ -243,214 +222,193 @@ fclose_and_clear(FILE **fpp)
+    set_core(false);
+#endif
 
-    lock_writelock(session, &g.prepare_commit_lock);
-    testutil_check(conn->set_timestamp(conn, buf));
-    lock_writeunlock(session, &g.prepare_commit_lock);
-    trace_msg(
-      "%-10s oldest=%" PRIu64 ", stable=%" PRIu64, "setts", g.oldest_timestamp, g.stable_timestamp);
+    WT_UNUSED(cursor);
+    WT_UNUSED(tag);
 }
 
 /*
- * timestamp --
- *     Periodically update the oldest timestamp.
+ * table_dump_page --
+ *     Dump a page from a table (at a given key number) to a backing file.
  */
-WT_THREAD_RET
-timestamp(void *arg)
+void
+table_dump_page(
+  WT_SESSION *session, const char *checkpoint, TABLE *tbl, uint64_t keyno, const char *tag)
 {
-    WT_CONNECTION *conn;
-    WT_SESSION *session;
-
-    (void)arg; /* Unused argument */
-
-    conn = g.wts_conn;
-
-    /* Locks need session */
-    testutil_check(conn->open_session(conn, NULL, NULL, &session));
-
-    /* Update the oldest and stable timestamps at least once every 15 seconds. */
-    while (!g.workers_finished) {
-        random_sleep(&g.rnd, 15);
-
-        timestamp_once(session, true, false);
+    WT_CURSOR *cursor;
+    WT_ITEM key;
+    int exactp, ret;
+    char cfg[256];
+
+    if (checkpoint != NULL)
+        testutil_check(__wt_snprintf(cfg, sizeof(cfg), "checkpoint=%s", checkpoint));
+
+    wt_wrap_open_cursor(session, tbl->uri, checkpoint == NULL ? NULL : cfg, &cursor);
+    switch (tbl->type) {
+    case FIX:
+    case VAR:
+        cursor->set_key(cursor, keyno);
+        break;
+    case ROW:
+        key_gen_init(&key);
+        key_gen(tbl, &key, keyno);
+        cursor->set_key(cursor, &key);
+        key_gen_teardown(&key);
+        break;
     }
-
-    testutil_check(session->close(session, NULL));
-    return (WT_THREAD_RET_VALUE);
+    ret = cursor->search_near(cursor, &exactp);
+    if (ret == 0)
+        cursor_dump_page(cursor, tag);
+    else
+        fprintf(stderr, "%s: Not dumping (error %d from search_near)\n", tag, ret);
+    testutil_check(cursor->close(cursor));
 }
 
 /*
- * timestamp_teardown --
- *     Wrap up timestamp operations.
+ * set_core --
+ *     Turn core dumps off/on.
  */
 void
-timestamp_teardown(WT_SESSION *session)
+set_core(bool off)
 {
+#ifdef HAVE_SETRLIMIT
+    static bool saved = false;
+    static struct rlimit saved_rlim;
+    struct rlimit rlim;
+
     /*
-     * Do a final bump of the oldest and stable timestamps, otherwise recent operations can prevent
-     * verify from running.
+     * This could race if a lot of threads failed at the same time, but it's unlikely (and
+     * unimportant) enough that I'm not fixing it.
      */
-    timestamp_once(session, false, true);
+    if (!saved) {
+        testutil_assert_errno(getrlimit(RLIMIT_CORE, &saved_rlim) == 0);
+        saved = true;
+    }
+    rlim = saved_rlim;
+    if (off)
+        rlim.rlim_cur = 0;
+    testutil_assert_errno(setrlimit(RLIMIT_CORE, &rlim) == 0);
+#endif
```

Comment:
I'm not sure about when we don't have `rlimit()` (Windows?) but we should probably have a `WT_UNUSED()` for `off` in that case.

Reply:
I thought test format is not supposed to run on Windows?

Reply:
OK. It might be some other Unix-like thing then. I realise this code was just moved from elsewhere but it's a two-line change.